### PR TITLE
[CSL-1586] /healthcheck/route53 endpoint

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
This single-commit PR is essentially the same as https://github.com/input-output-hk/cardano-sl/pull/1552 but rebased against `1.0` and with the port being configurable.

We default to `3030` (taken arbitrarily) and `localhost` if no argument is passed on the command line, to smooth out deployments. Happy to take suggestions if the above behaviour is not desirable.

### CODE REVIEW

**Comment by adinapoli-iohk** [comment]
@domenkozar Oh! Appveyor is failing it seems:

```
--  While building package node-sketch-0.2.0.0 using:
      C:\s\setup-exe-cache\x86_64-windows\Cabal-simple_Z6RU0evB_1.24.2.0_ghc-8.0.2.exe --builddir=.w\dist\ca59d0ab build lib:node-sketch exe:discovery exe:ping-pong --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: C:\w\.w\logs\node-sketch-0.2.0.0.log
    Preprocessing library node-sketch-0.2.0.0...
    <command line>: cannot satisfy -package-id statistics-0.13.3.0-vN6s2qb86Z5lmqDDU9KzO
        (use -v for more information)
Sorry, we tried running command for 5 times and all attempts were unsuccessful!
```

Is there any hope of restarting a build there? I couldn't find any button, but maybe I haven't looked hard enough 😁

**Comment by domenkozar** [comment]
Done

File: node/src/Pos/HealthCheck/Route53.hs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import qualified Options.Applicative    as Opt
+import           Pos.Util.TimeWarp      (NetworkAddress, addrParserNoWildcard)
+import           Serokell.Util.OptParse (fromParsec)
+import           Universum
+
+data Route53HealthCheckParams = Route53HealthCheckParams
+    { route53HealthCheckHost :: !ByteString
+    , route53HealthCheckPort :: !Int
+    } deriving Show
```
Comment (dcoutts) [inline node/src/Pos/HealthCheck/Route53.hs]: Do we need this? Won't `NetworkAddress` do?

File: node/src/Pos/HealthCheck/Route53.hs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+route53HealthCheckOption :: Opt.Parser Route53HealthCheckParams
+route53HealthCheckOption = do
+    addr <- route53HealthCheckServerOption
+    pure $ Route53HealthCheckParams
+        { route53HealthCheckHost = fst addr
+        , route53HealthCheckPort = fromIntegral (snd addr)
+        }
+
+route53HealthCheckServerOption :: Opt.Parser NetworkAddress
+route53HealthCheckServerOption = Opt.option (fromParsec addrParserNoWildcard) $
```
Comment (dcoutts) [inline node/src/Pos/HealthCheck/Route53.hs]: Why `addrParserNoWildcard` rather than `addrParser`?

File: node/src/Pos/Launcher/Param.hs [hunk 3]
```diff
@@ -55,23 +56,24 @@ data TransportParams = TransportParams
...
+    , npSecretKey      :: !SecretKey                        -- ^ Primary secret key of node
+    , npUserSecret     :: !UserSecret                       -- ^ All node secret keys
+    , npBaseParams     :: !BaseParams                       -- ^ See 'BaseParams'
+    , npGenesisCtx     :: !GenesisContext                   -- ^ Predefined genesis context
+    , npJLFile         :: !(Maybe FilePath)                 -- TODO COMMENT
+    , npReportServers  :: ![Text]                           -- ^ List of report server URLs
+    , npUpdateParams   :: !UpdateParams                     -- ^ Params for update system
+    , npUseNTP         :: !Bool                             -- ^ Whether to use synchronisation with NTP servers.
+    , npTransport      :: !TransportParams                  -- ^ (TCP) transport parameters.
+    , npRoute53Params  :: !(Maybe Route53HealthCheckParams) -- ^ Where to listen for the Route53 DNS health-check.
```
Comment (dcoutts) [inline node/src/Pos/Launcher/Param.hs]: If this were just `NetworkAddress` we also wouldn't need to change all the layout :-)

**Comment by adinapoli-iohk** [comment]
Done.

**Pull request merged**